### PR TITLE
feat(parser/html): parse doctype declarations

### PR DESCRIPTION
**Pull request by dyc3**
## Summary

This PR completes everything necessary to parse all valid doctype directives.

> [!CAUTION]
> This PR is stacked on the following PRs. **Do not merge it without merging the others and rebasing this branch.** To review this PR, look at the last commit only.
> - #3951

## Test Plan

Added tests.

### CODE REVIEW

**Review by chansuke — APPROVED** [review]
it seems nice to me

File: crates/biome_html_parser/src/lexer/mod.rs [hunk 1]
```diff
@@ -233,6 +253,41 @@ impl<'src> HtmlLexer<'src> {
...
+                }
+                _ if current.is_ascii() => {
+                    self.advance(1);
+                    content_started = true;
+                }
+                _ => break,
+            }
+        }
+
+        if content_started && !encountered_invalid {
```
Comment (arendjr) [inline crates/biome_html_parser/src/lexer/mod.rs:279]: Is it allowed for an unquoted attribute value to be empty? We might want to handle that case.

**Review by arendjr — APPROVED** [review]
LGTM. Just one comment.

File: crates/biome_html_parser/src/lexer/mod.rs (hunk 1 above)
Comment (arendjr) [reply crates/biome_html_parser/src/lexer/mod.rs:279]: Nevermind, this belonged with the other PR.

**Pull request merged**